### PR TITLE
e2e GCSMove Test Scenarios

### PR DESCRIPTION
**Pull request by Swati10-11**
@itsankit-google @rmstar  Please review

cc @sawantpritam

### CODE REVIEW

File: src/e2e-test/features/gcsmove/GCSMove.feature [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  @GCS_READ_RECURSIVE_TEST @GCS_SINK_TEST
+  Scenario:Validate successful move objects from one bucket to another with Move All Subdirectories set to true
+    Given Open Datafusion Project to configure pipeline
+    When Expand Plugin group in the LHS plugins list: "Conditions and Actions"
+    When Select plugin: "GCS Move" from the plugins list as: "Action"
+    When Navigate to the properties page of plugin: "GCS Move"
+    Then Enter GCSMove property projectId "projectId"
+    Then Enter GCSMove property source path "gcsMoveReadRecursivePath"
+    Then Enter GCSMove property destination path
+    Then Select GCSMove property move all subdirectories as "true"
```
Comment (rmstar) [inline src/e2e-test/features/gcsmove/GCSMove.feature:33]: Test with more than 1 subdirectory. Also test move all subdirectories as "false"

**Review by rmstar — COMMENTED** [review]
Add test for auto create location set to non-default value. See https://cloud.google.com/storage/docs/locations for available locations

File: src/e2e-test/features/gcsmove/GCSMove.feature (hunk 1 above)
Comment (Swati10-11) [reply src/e2e-test/features/gcsmove/GCSMove.feature:33]: test move all subdirectories as "false"
Added the test

Comment (Swati10-11) [reply src/e2e-test/features/gcsmove/GCSMove.feature:33]: Add test for auto create location set to non-default value
Added the test

File: src/e2e-test/java/io/cdap/plugin/gcsmove/stepsdesign/GCSMove.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                Assert.fail("Target gcs bucket " + targetGCSBucket + " not created - " + e.getMessage());
+            } else {
+                Assert.fail(e.getMessage());
+            }
+        }
+    }
+
+    @Then("Validate GCSMove failed to move object {string} to destination bucket")
+    public void validateGCSMoveFailedToMoveObjectToDestinationBucket(String path) throws IOException {
+        String sourceGCSBucket = TestSetupHooks.gcsSourceBucketName;
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/gcsmove/stepsdesign/GCSMove.java]: indent is incorrect. please reformat all files in the PR using these settings - https://cdap.atlassian.net/wiki/spaces/CE/pages/1594098619/Coding+Standard

File: src/e2e-test/features/gcsmove/GCSMove.feature [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  @GCS_CSV_TEST @GCS_SINK_TEST
+  Scenario:Validate successful move object from one bucket to another new bucket with location set to non-default value
+    Given Open Datafusion Project to configure pipeline
+    When Expand Plugin group in the LHS plugins list: "Conditions and Actions"
+    When Select plugin: "GCS Move" from the plugins list as: "Action"
+    When Navigate to the properties page of plugin: "GCS Move"
+    Then Enter GCSMove property projectId "projectId"
+    Then Enter GCSMove property source path "gcsCsvFile"
+    Then Enter GCSMove property destination path
+    Then Enter GCSMove property location "locationEU"
```
Comment (rmstar) [inline src/e2e-test/features/gcsmove/GCSMove.feature:109]: can we verify that the destination bucket is created in the specified location?

Comment (Swati10-11) [reply src/e2e-test/features/gcsmove/GCSMove.feature:109]: Changes done

File: src/e2e-test/java/io/cdap/plugin/gcsmove/stepsdesign/GCSMove.java (hunk 2 above)
Comment (Swati10-11) [reply src/e2e-test/java/io/cdap/plugin/gcsmove/stepsdesign/GCSMove.java]: Reformat all files done

**Review by rmstar — APPROVED** [review]
please squash commits

**Comment by rmstar** [comment]
Please resolve conflicts

**Comment by Swati10-11** [comment]
squashed commits and resolved conflicts

**Comment by rmstar** [comment]
Please take a look at the test failures

**Comment by Swati10-11** [comment]
> Please take a look at the test failures
Tests are failing with sandbox OOM issue     https://cdap.atlassian.net/browse/CDAP-19026

**Comment by Swati10-11** [comment]
@rmstar  "Sandbox OOM issue is resolved. Commits Squashed and conflicts resolved"

**Comment by itsankit-google** [comment]
Please resolve conflicts

**Comment by Swati10-11** [comment]
@itsankit-google  "Sandbox OOM issue is resolved. Commits Squashed and conflicts resolved"

**Comment by Swati10-11** [comment]
> Please resolve conflicts

@itsankit-google "Sandbox OOM issue is resolved. Commits Squashed and conflicts resolved"

**Pull request merged**